src/kernel/arch/x86/paging.zig

```diff
@@ -196,3 +204,28 @@ test "mapDir" {
     checkDirEntry(dir.entries[virt_start / PAGE_SIZE + 1], phys_start + PAGE_SIZE);
     expectEqual(dir.entries[virt_start / PAGE_SIZE + 2], 0);
 }
+
+// The label to jump to after causing a page fault. This is needed as we don't want to cause an
+// infinite loop by jummping to the same instruction that caused the fault.
+extern var rt_fault_callback: *u32;
+
+var faulted = false;
+
+fn testPageFault(ctx: *arch.InterruptContext) void {
+    faulted = true;
+    // Return to the fault callback
+    ctx.eip = @ptrToInt(&rt_fault_callback);
+}
+
+fn runtimeTests(v_end: u32) void {
```

Comment:
You could add a test that when accessing memory within a valid page doesn't cause a page fault. So just copy the runtimeTests function and change the pointer address and change the assert to `expect(!faulted)`

Reply:
I didn't add a test for accessing mapped memory since if it didn't work then we wouldn't even get past the point of setting the page directory without faulting xD

Reply:
Good point lol. I was thinking maybe it is best not to write to a random location as if successes, then could cause problems if the memory was important. Maybe just read.

Reply:
Agreed :). On second thoughts, maybe I should add an explicit test for reading mapped memory, since if we change it so that a page fault isn't always fatal then this test may be reached even if a fault occurred.